src/csv.rs

```diff
@@ -41,8 +41,11 @@ impl Row {
 }
 
 impl CsvLensReader {
-    pub fn new(filename: &str) -> Result<Self> {
-        let mut reader = Reader::from_path(filename)?;
+    pub fn new(filename: &str, delimiter: Option<u8>) -> Result<Self> {
+        let mut reader = match delimiter {
+            Some(d) => ReaderBuilder::new().delimiter(d).from_path(filename)?,
```

Comment:
There are a few other places where a csv reader is constructed (`csv::ReaderInternalState` and `find::FinderInternalState`) - they need to have access to this parameter also. When testing using this branch the behavior of searching for a string is incorrect and it's probably related to this.

I also wonder if there is a cleaner way to pass this and potentially other configurations around to all these places.